pkgs/development/compilers/dart/package-source-builders/printing/printing.patch

```diff
@@ -3,15 +3,15 @@
 @@ -16,6 +16,7 @@
  set(PROJECT_NAME "printing")
  project(${PROJECT_NAME} LANGUAGES CXX)
- 
-+set(PDFIUM_DIR $ENV{PDFIUM_DIR})
```

Comment:
Doesn't setting `env.PDFIUM_DIR` / `cmakeFlags` work?

Reply:
After staring at CMake docs for a bit, you might be able to get away with it in a more CMake-friendly way by replacing the `download_project` call with a `FetchContent_Declare` then `FetchContent_MakeAvailable`:

```diff
-include(../windows/DownloadProject.cmake)
-download_project(PROJ
-                 pdfium
-                 URL
-                 ${PDFIUM_URL})
+include(FetchContent)
+FetchContent_Declare(
+    pdfium
+    URL ${PDFIUM_URL}
+    URL_HASH <some-hash>
+)
+FetchContent_MakeAvailable(pdfium)
```

Then set these CMake flags:
```nix
  cmakeFlags = [
    (lib.cmakeBool "FETCHCONTENT_FULLY_DISCONNECTED" true)
    (lib.cmakeFeature "FETCHCONTENT_SOURCE_DIR_PDFIUM" "${pdfium-binaries}")
  ];
```

Obviously, this is a lot of work, but it's technically less hacky ;)

Reply:
Interesting... I might have misunderstood how Dart compiles these packages — these derivations are only supposed to provide patched *sources* to Dart rather than be individually built. In that case I think your approach works fine, though maybe using `pkgs.replaceVars` on the patch file would be terser:

```diff
# printing/printing.patch
-set(PDFIUM_DIR CUSTOM_PDFIUM_DIR)
+set(PDFIUM_DIR @pdfium@)
```

```diff
# printing/default.nix
-  patches = [ ./printing.patch ];
+  patches = [ 
+    (replaceVars ./printing.patch {
+      pdfium = pdfium-binaries;
+    })
+  };
```